### DISCUSSION TITLE
Help with Backrest retention policy

### DISCUSSION THREAD

**Discussion by malavolti** [garethgeorge/backrest#445] [Q&A]
Thank you Gareth for your beautiful work,

by reading [the documentation](https://restic.readthedocs.io/en/latest/060_forget.html) I've found:

    --keep-within duration keep all snapshots having a timestamp within the specified duration of the latest snapshot, where duration is a number of years, months, days, and hours. E.g. 2y5m7d3h will keep all snapshots made in the two years, five months, seven days and three hours before the latest (most recent) snapshot.

If I want to keep all snapshot with duration '1y6m4w7d7h', what I have to do on my Backrest?

I've already set as the image shows:

![Retention-Backrest](https://github.com/user-attachments/assets/74947aa6-a2f1-4eb7-a92e-b376d1bcc6d1)

It it means that I delete all snapshots that are older than 1y6m4w7d24h or other? 

I get this from backrest:

![backrest-backup-retention](https://github.com/user-attachments/assets/8f6bc3c1-c9ef-4368-9421-e632a9804708)

Thank you to anybody that want to help me to understand it! 
Marco

**Comment by garethgeorge** [comment]
Hey, glad backrest is still working well for you :) 

backrest doesn’t actually support the keep by duration flag at the moment. 

The time bucketed retention policy is essentially looping over snapshots, sorting them into buckets by each configured window, and then removing entries older than the limit for each bucket. 


E.g. buckets of daily 24, and weekly 12

Will keep a choice of one snapshot for each hour in the last 24 and it will keep a choice of 1 snapshot in each week for the last 12 weeks.

**Reply by malavolti** [reply]
Thank you @garethgeorge. Do you have any Retention Policy to suggest for a good backup retention?

**Reply by garethgeorge** [reply]
Hey! I personally use the default retention for most of my plans. 

On a few machines the other retention policy I use is 48 hours and then 90 days when I want a lot of versions.
